### PR TITLE
Fix #984 introduce the CamelContextCustomizerBuildItem

### LINKED ISSUE #984: Introduce the CamelContextCustomizerBuildItem to do further customization of the camel context
This idea comes up during the discussion of https://github.com/apache/camel-quarkus/issues/819

CamelContextCustomizerBuildItem  could be used to add the service in the camel context when we have the LRASagaService just like the auto configuration in the spring boot.

It needs a proper class level JavaDoc to explain:

- What happens before and after
- Sum up roughly which uses is it intended for and
- Explain the relationship to Core and Main.

### PR DESCRIPTION
**Pull request by zhfeng**
Fix https://github.com/apache/camel-quarkus/issues/984

### CODE REVIEW

**Comment by lburgazzoli** [comment]
Added some comment, in addition it would be nice to have some tests

**Comment by zhfeng** [comment]
Thanks @lburgazzoli - I'm working on the tests and javadoc. But I can not see your comments, is there anything wrong ?

**Comment by lburgazzoli** [comment]
> But I can not see your comments, is there anything wrong ?

I left the review pending sorry, they should be visible now

File: extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextCustomizerBuildItem.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.core.deployment;
+
+import io.quarkus.builder.item.SimpleBuildItem;
+import io.quarkus.runtime.RuntimeValue;
+import org.apache.camel.quarkus.core.CamelContextCustomizer;
+
+public class CamelContextCustomizerBuildItem extends SimpleBuildItem {
```
Comment (ppalaga) [inline extensions-core/core/deployment/src/main/java/org/apache/camel/quarkus/core/deployment/CamelContextCustomizerBuildItem.java:23]: Please add a short class level javadoc informing about the purpose of the builditem.

File: extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextCustomizer.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.camel.quarkus.core;
+
+import org.apache.camel.CamelContext;
+
+public interface CamelContextCustomizer {
```
[the remark this answers is not in the document]
Comment (zhfeng) [reply extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/CamelContextCustomizer.java:22]: yeah, I think it makes sense.

**Comment by zhfeng** [comment]
@lburgazzoli I made the changes as your suggestions and add the unit test of the CamelContextCustomizer.

**Pull request merged**